src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,131 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int BASIC_LENGTH = 10;
+    private T[] list;
+    private int size;
+
+    public ArrayList() {
+        list = (T[]) new Object[BASIC_LENGTH];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == list.length) {
+            addSpase();
+        }
+        list[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Incorrect index of list");
+        } else {
```

Comment:
you can simplify this and remove the `else` line (do not remove the body of `else` clause.